File: app/schemas/pharmacy.py
from sqlmodel import SQLModel
from typing import List
from datetime import datetime

class PresItem(SQLModel):
    medicine_id: int
    quantity: int
    usage: str

class PrescriptionCreate(SQLModel):
    record_id: int
    items: List[PresItem]

# 专门用于返回的 Schema，防止 Swagger 解析报错
class PrescriptionRead(SQLModel):
    pres_id: int
    record_id: int
    total_amount: float
    create_time: datetime


class MedicinePurchase(SQLModel):
    medicine_id: int
    quantity: int


class MedicineCreate(SQLModel):
    name: str
    price: float
    stock: int
    unit: str


class UsagePoint(SQLModel):
    date: str
    quantity: int


class MedicineInventory(SQLModel):
    medicine_id: int
    name: str
    price: float
    stock: int
    unit: str
    usage_30d: int
    avg_daily_usage: float
    expected_week_usage: int
    suggested_restock: int
    needs_restock: bool
    usage_trend: List[UsagePoint]

File: frontend/src/api/modules/pharmacy.ts
import http from "../http";

export interface UsagePoint {
  date: string;
  quantity: number;
}

export interface MedicineItem {
  medicine_id: number;
  name: string;
  price: number;
  stock: number;
  unit: string;
  usage_30d: number;
  avg_daily_usage: number;
  expected_week_usage: number;
  suggested_restock: number;
  needs_restock: boolean;
  usage_trend: UsagePoint[];
  usage_monthly: UsagePoint[];
}

export interface PrescriptionItemPayload {
  medicine_id: number;
  quantity: number;
  usage: string;
}

export interface PrescriptionPayload {
  record_id: number;
  items: PrescriptionItemPayload[];
}

export interface MedicinePurchasePayload {
  medicine_id: number;
  quantity: number;
}

export interface MedicineCreatePayload {
  name: string;
  price: number;
  stock: number;
  unit: string;
}

export function fetchMedicines() {
  return http.get<MedicineItem[]>("/api/pharmacy/medicines");
}

export function createPrescription(payload: PrescriptionPayload) {
  return http.post("/api/pharmacy/prescriptions", payload);
}

export function fetchPrescriptionByRecord(recordId: number) {
  return http.get(`/api/pharmacy/prescriptions/by_record/${recordId}`);
}

export function purchaseMedicine(payload: MedicinePurchasePayload) {
  return http.post<MedicineItem>("/api/pharmacy/medicines/purchase", payload);
}

export function createMedicine(payload: MedicineCreatePayload) {
  return http.post<MedicineItem>("/api/pharmacy/medicines", payload);
}

export function replenishMedicines() {
  return http.post<MedicineItem[]>("/api/pharmacy/medicines/replenish");
}
